src/kirby/components/icon/custom-icon-name.directive.tns.ts

```diff
@@ -0,0 +1,65 @@
+import { LayoutBase } from 'tns-core-modules/ui/layouts/layout-base';
+import { Label } from 'tns-core-modules/ui/label/label';
+import {
+  Directive,
+  ElementRef,
+  Input,
+  Inject,
+  AfterViewInit,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+
+import { CUSTOM_FONT_SETTINGS, CustomIconSettings } from './custom-icon-settings';
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: '[customName]',
+})
+export class CustomIconNameDirective implements AfterViewInit, OnChanges {
+  // tslint:disable-next-line:no-input-rename
+  @Input('customName') customName: string;
+
+  constructor(
+    private element: ElementRef,
+    @Inject(CUSTOM_FONT_SETTINGS) private customIconSettings: CustomIconSettings
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.handleCustomIcon(this.customName);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.customName) {
+      this.handleCustomIcon(this.customName);
+    }
+  }
+
+  handleCustomIcon(name: string) {
+    if (name && this.customIconSettings[0].icons) {
```

Comment:
what is the `[0]` good for here? `customIconSettings` is not an array.

Or what am i missing here 😄

Reply:
It is an array because 'multi: true' under '{ provide: CUSTOM_FONT_SETTINGS, useValue: customIconSettings, multi: true },' if it's set to false it will not be an array.

Reply:
Support for having multiple custom fonts is removed.